### DISCUSSION TITLE
Changing refetchInterval for a query.

### DISCUSSION THREAD

**Discussion by nanandn** [TanStack/query#1460] [General]
Is it possible to change the refetchInterval for a query based on the response it returns. I have a query that fetches status at regular interval. Initially the refetchInterval is 5 minutes. But I want to change this interval if the status return by this query changes to say "IN_PROGRESS". 

```
useQuery("getStatus", fetchStatus, {
    refetchInterval: 1000*60*5,
    onSuccess: ({status}) => {
           if (prevStatus !== "IN_PROGRESS" && status === "IN_PROGRESS") {
                // Here I want to change the interval of subsequent re-fetches to 30 seconds.
           } else if (prevStatus === "IN_PROGRESS" && status !== "IN_PROGRESS") {
               // Change the refetchInterval back to 5 minutes.
           }
    }
});
```

**Comment by TkDodo** [comment]
Yes, it is. You need to keep some local state that holds the interval value and update that state in `onSuccess`.

**Reply by nanandn** [reply]
@TkDodo Thanks. That's what I ended up doing. Seems to be working. I was wondering if there is anything like query.setOptions kind of API to do that instead of keeping another state.
